astropy/table/column.py

```diff
@@ -697,7 +697,7 @@ def quantity(self):
         # the Quantity initializer is used here because it correctly fails
         # if the column's values are non-numeric (like strings), while .view
         # will happily return a quantity with gibberish for numerical values
-        return Quantity(self, copy=False, dtype=self.dtype, order='A')
+        return Quantity(self, self.unit, copy=False, dtype=self.dtype, order='A', subok=True)
```

Comment:
I was a bit surprised that you needed to pass `self.unit` here, and feel that is a bug in `Quantity`. But probably best addressed separately - see #8430.

Anyway, this is the right way to do it.